src/redux/selectors/dappModalsSelectors.ts

```diff
@@ -0,0 +1,34 @@
+import { RootState } from '../store';
+import { createDeepEqualSelector } from './helpers';
+
+export const modalsSliceSelector = (state: RootState) => state.dappModal;
+
+export const dappModalVisibilitySelector = createDeepEqualSelector(
+  modalsSliceSelector,
+  (state) => state.visible
+);
+
+export const dappModalHeaderVisibilitySelector = createDeepEqualSelector(
+  modalsSliceSelector,
```

Comment:
I think you should create an extra selector for state.modal and use that in the lower selectors, so you can just do state.showHeader, state.showFooter etc